### PR TITLE
21 clean code

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

File: pkg/password/fyne.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+// Ask displays a password dialog using the given nickname.
+//
+// It returns the entered password and any error that may have occurred.
+// Note: Reuses the Fyne wrapped application.
+func (f *FynePrompter) Ask(name string) (string, error) {
+	result := <-PasswordDialog(name, f.guiApp)
+	return result.password, result.err
+}
+
+// NewFynePrompter creates a new password rpomter with the given Fyne GUI application.
```
Comment (SebastienValla) [inline pkg/password/fyne.go:29]:
```suggestion
// NewFynePrompter creates a new password prompter with the given Fyne GUI application.
```

File: pkg/wallet/wallet.go [hunk 2]
```diff
@@ -7,50 +7,43 @@ import (
...
-type Config struct {
-	// address
-	Wallets []KeyPair `json:"wallets"`
+	Version  uint8
+	Nickname string
+	Address  string
+	KeyPair  KeyPair
 }
 
-func (w *Wallet) Protect(password string, keyPairIndex uint8) error {
```
Comment (SebastienValla) [inline pkg/wallet/wallet.go:52]: Protect and unProtect function contains duplicated code that could be factorized

File: pkg/wallet/wallet.go [hunk 3]
```diff
@@ -151,29 +143,34 @@ func LoadAll() (wallets []Wallet, e error) {
 	return wallets, nil
 }
 
+// Load loads the wallet taht match the given name in the working directory
```
Comment (SebastienValla) [inline pkg/wallet/wallet.go:146]:
```suggestion
// Load loads the wallet that match the given name in the working directory
```

File: pkg/wallet/wallet.go [hunk 4]
```diff
@@ -151,29 +143,34 @@ func LoadAll() (wallets []Wallet, e error) {
...
-		return nil, fmt.Errorf("unmarshaling file 'wallet_%s.json': %w", nickname, err)
+		return nil, fmt.Errorf("unmarshalling file '%s': %w", walletName, err)
 	}
 
 	return &wallet, nil
 }
 
-func New(nickname string) (*Wallet, error) {
-	pubKey, privKey, err := ed25519.GenerateKey(nil)
+// Generate instanciates a new wallet, protects its private key and persists it.
```
Comment (SebastienValla) [inline pkg/wallet/wallet.go:165]:
```suggestion
// Generate instantiates a new wallet, protects its private key and persists it.
```

File: pkg/wallet/wallet.go [hunk 5]
```diff
@@ -189,40 +186,53 @@ func New(nickname string) (*Wallet, error) {
...
-		Address:  "A" + base58.CheckEncode(append(make([]byte, 1), addr[:]...)),
-		KeyPairs: []KeyPair{{
-			PrivateKey: privKey,
-			PublicKey:  pubKey,
-			Salt:       salt,
-			Nonce:      nonce,
-		}},
+	wallet, err := New(nickname, addr, privateKey, publicKey, salt, nonce)
+	if err != nil {
+		return nil, fmt.Errorf("instanciating a new wallet: %w", err)
```
Comment (SebastienValla) [inline pkg/wallet/wallet.go:191]:
```suggestion
		return nil, fmt.Errorf("instantiating a new wallet: %w", err)
```

File: pkg/wallet/wallet.go [hunk 6]
```diff
@@ -189,40 +186,53 @@ func New(nickname string) (*Wallet, error) {
...
 	if err != nil {
-		return nil, fmt.Errorf("writing wallet to 'wallet_%s.json': %w", nickname, err)
+		return nil, fmt.Errorf("persisting the new wallet: %w", err)
+	}
+
+	return wallet, nil
+
+}
+
+// New instanciates a new wallet.
```
Comment (SebastienValla) [inline pkg/wallet/wallet.go:208]:
```suggestion
// New instantiates a new wallet.
```

**Review by SebastienValla — CHANGES_REQUESTED** [review]

**Comment by gregLibert** [comment]
@SebastienValla correction pushed

**Review by SebastienValla — APPROVED** [review]
great !

File: internal/handler/html/html.go [hunk 7]
```diff
@@ -34,6 +34,13 @@ func WebWalletHandler(params operations.WebParams) middleware.Responder {
 	return helper.NewCustomResponder(resourceContent, header, http.StatusOK)
 }
 
-func DefaultPageHandler(_ operations.DefaultPageParams) middleware.Responder {
+// DefaultRedirectHandler redirects request to "/" URL to "web/index.html"
+func DefaultRedirectHandler(_ operations.DefaultPageParams) middleware.Responder {
 	return helper.NewCustomResponder(nil, map[string]string{"Location": "web/index.html"}, http.StatusPermanentRedirect)
 }
+
+// AppendEndpoints append web endpoints to the API.
```
Comment (thomas-senechal) [inline internal/handler/html/html.go:42]:
```suggestion
// AppendEndpoints appends web endpoints to the API.
```

**Review by thomas-senechal — APPROVED** [review]

File: internal/handler/html/html.go (hunk 7 above)
Comment (gregLibert) [reply internal/handler/html/html.go]: fixed

**Pull request merged**